### PR TITLE
PR: update Intune assignment logic - Fixes 3892 and 3921

### PR DESCRIPTION
**Pull request by William-Francillette**
#### This Pull Request (PR) fixes the following issues

* M365DSCDRGUtil
  * Added ConvertFrom-IntunePolicyAssignment and ConvertTo-IntunePolicyAssignment
  FIXES [#3892](https://github.com/microsoft/Microsoft365DSC/issues/3892)
* IntuneWindowsAutopilotDeploymentProfileAzureADJoined
  * Modified assigned to use sdk instead of API call and added logic to use groupDisplayName in assignment
  FIXES [#3892](https://github.com/microsoft/Microsoft365DSC/issues/3892)
* IntuneWindowsAutopilotDeploymentProfileAzureADHybridJoined
  * Modified assigned to use sdk instead of API call and added logic to use groupDisplayName in assignment
  FIXES [#3892](https://github.com/microsoft/Microsoft365DSC/issues/3892)
* IntuneWindowsUpdateForBusinessRingUpdateProfileWindows10
  * Modified assigned to use sdk instead of API call and added logic to use groupDisplayName in assignment
  FIXES [#3892](https://github.com/microsoft/Microsoft365DSC/issues/3892)
* IntuneDeviceConfigurationPolicyWindows10
  * Modified assigned to use sdk instead of API call and added logic to use groupDisplayName in assignment
  FIXES [#3921](https://github.com/microsoft/Microsoft365DSC/issues/3921)
* IntuneDeviceEnrollmentStatusPageWindows10
  * Fixed assignments using API call
  FIXES [#3921](https://github.com/microsoft/Microsoft365DSC/issues/3921)

### CODE REVIEW

File: Modules/Microsoft365DSC/Modules/M365DSCDRGUtil.psm1 [hunk 1]
```diff
@@ -1160,7 +1272,7 @@ function Update-DeviceConfigurationPolicyAssignment
             $deviceManagementPolicyAssignments += @{'target' = $formattedTarget}
         }
         $body = @{'assignments' = $deviceManagementPolicyAssignments} | ConvertTo-Json -Depth 20
-        #write-verbose -Message $body
+        write-verbose -Message $body
```
Comment (ykuijs) [inline Modules/Microsoft365DSC/Modules/M365DSCDRGUtil.psm1:1275]: I think this is still here for troubleshooting and should be commented out. Correct?

**Review by ykuijs — COMMENTED** [review]
One small comment

**Comment by ykuijs** [comment]
@William-Francillette Unit tests are failing. Could you please have a look and fix them?

File: Modules/Microsoft365DSC/Modules/M365DSCDRGUtil.psm1 (hunk 1 above)
Comment (William-Francillette) [reply Modules/Microsoft365DSC/Modules/M365DSCDRGUtil.psm1]: Removed debugging trace

**Comment by William-Francillette** [comment]
> @William-Francillette Unit tests are failing. Could you please have a look and fix them?

@ykuijs all sorted now

**Review by ykuijs — APPROVED** [review]
LGTM

**Comment by Cyanic-Cloud** [comment]
@William-Francillette 

**Quick one, will any of the upcoming fixes help with configuration drift failing due to policy ID number?**

See below Event Viewer log:

`
<M365DSCEvent>
    <ConfigurationDrift Source="MSFT_IntuneDeviceConfigurationHealthMonitoringConfigurationPolicyWindows10">
        <ParametersNotInDesiredState>
            <Param Name="Id"><CurrentValue>f85e6911-73ae-4be6-ac69-0a567e2720e3</CurrentValue><DesiredValue>7ddcdcc9-b566-4805-a5e6-d4f2299131bd</DesiredValue></Param>
        </ParametersNotInDesiredState>
    </ConfigurationDrift>
    <DesiredValues>
`
    
    
    
    **DSC Config Below:**
    
            IntuneDeviceConfigurationHealthMonitoringConfigurationPolicyWindows10 "IntuneDeviceConfigurationHealthMonitoringConfigurationPolicyWindows10-Intune data collection policy"
        {
            AllowDeviceHealthMonitoring       = "enabled";
            ApplicationId                     = $ApplicationId
            Assignments                       = @(
                MSFT_DeviceManagementConfigurationPolicyAssignments{
                    deviceAndAppManagementAssignmentFilterType = 'none'
                    dataType = '#microsoft.graph.allDevicesAssignmentTarget'
                });
            CertificateThumbprint             = $Thumbprint
            ConfigDeviceHealthMonitoringScope = @("bootPerformance");
            Description                       = "";
            DisplayName                       = "Intune data collection policy";
            Ensure                            = "Present";
            Id                                = "7ddcdcc9-b566-4805-a5e6-d4f2299131bd";
            SupportsScopeTags                 = $True;
            TenantId                          = $TenantId
        }
        
        Its failing on ID, should this be the case?

**Comment by William-Francillette** [comment]
@Cyanic-Cloud #3893 will fix the id issue
If you find anything else don't hesitate to raise an issue and will look at it

Thanks

**Pull request merged**